File: weel_02/yumyum/stack.js
const stack = [];
let top = 0;
let answer = "";

const commands = {
  push: (x) => (stack[top++] = x),
  pop: () => {
    if (top) {
      top--;
      return stack.splice(-1);
    } else return -1;
  },
  size: () => top,
  empty: () => (!top ? 1 : 0),
  top: () => (top ? stack[top - 1] : -1),
};

const readline = require("readline");
const line = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let input = [];

line
  .on("line", (li) => input.push(li))
  .on("close", () => {
    input.map((li, idx) => {
      if (idx === 0) return;
      const [cmd, num] = li.split(" ");
      if (cmd === "push") {
        commands[cmd](parseInt(num));
      } else answer += `${commands[cmd]()}\n`;
    });

    console.log(answer);
    process.exit();
  });

File: week_03/comet/stack.java
import java.io.*;
import java.util.ArrayList;

public class Main {
    static ArrayList<Integer> stack = new ArrayList<>();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            String[] input = br.readLine().split(" ");

            switch (input[0]) {
                case "push":
                    push(Integer.parseInt(input[1]));
                    break;
                case "pop":
                    bw.write(pop() + "\n");
                    break;
                case "size":
                    bw.write(size() + "\n");
                    break;
                case "empty":
                    bw.write(empty() + "\n");
                    break;
                case "top":
                    bw.write(top() + "\n");
            }
        }

        bw.flush();
        bw.close();
    }

    static void push(int x) {
        stack.add(x);
    }

    static int pop() {
        if (stack.size() == 0) return -1;
        return stack.remove(stack.size() - 1);
    }

    static int size() {
        return stack.size();
    }

    static int empty() {
        if (stack.isEmpty()) return 1;
        return 0;
    }

    static int top() {
        if (stack.isEmpty()) return -1;
        return stack.get(stack.size() - 1);
    }
}